### PR TITLE
Shore up logging behavior

### LINKED ISSUE #70: Default to silent mode; allow verbose option
All my scripty usages have silent mode on, but I dislike needing the configuration block for every one.

Per the unix guidelines, I think scripty should be silent by default, and instead have a verbose flag. (Dry run mode should probably then imply verbose.)

### PR DESCRIPTION
**Pull request by jasonkarns**
## Goals

1. align with Unix philosophy as a CLI tool
2. be as user friendly as possible on the CLI (piping, etc)
3. conform to npm's conventions as much as possible

## Proposal

1. All scripty output goes to `STDERR` at all times.

Any output that scripty itself reports is purely diagnostic by
definition. Only the output from the scripts that scripty _executes_
should be considered "core data" and thus only the script output should
go to STDOUT.

Benefits: scripty becomes pipeable by default because the wrapped script
output is alone on STDOUT; scripty's STDERR output can be redirected
and/or ignored.

2. Conform to Unix and GNU conventions: `silent` and `quiet` flags are
aliases; accept `verbose`, `dryRun` implies `INFO`

3. Leverage npm's example and have a single `logLevel` setting that is
impacted by silent/quiet/verbose/dry-run flags.

## Implementation

This PR introduces the following `LOG LEVELS`

- verbose
- info * (the default)
- warn
- error
- silent

When executing a script, the message noting which script is about to be
executed is info level.

The contents of the script are printed at verbose level.

Scripty warnings and errors correspond to warn/error, respectively.

Therefore, by default, scripty's only output is the preamble to the
script, noting the path of the script about to be executed. (This is
similar to npm's default output for run-script.)

To be user-friendly per unix/gnu guidelines, the silent/quiet/verbose
flags all map to the logLevel option.

The logLevel option takes precedence if provided. Otherwise, in case of
conflicting flags (e.g. both `silent` and `verbose`), the noisier flag
takes precedence.

## TODO or followup

- [x] Match npm's output prefix ( add the `> ` prefix for info output)
- [ ] colorize ? (especially error output)
- [x] docs
- [ ] inherit npm's log level by default
- [x] conventional prefixing of warnings/errors

closes #70

### CODE REVIEW

**Comment by searls** [comment]
Woah, nice

File: lib/log.test.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      td.verify(console.error('mation'))
+    },
+    warn: function () {
+      subject.level(LEVEL.WARN)
+      td.when(console.error('mation')).thenThrow(new Error('Should not log info calls at WARN level'))
+      subject.info('mation')
+      subject.warn('ing')
+      td.verify(console.error('ing'))
+    },
+    error: function () {
```
[the remark this answers is not in the document]
Comment (jasonkarns) [reply lib/log.test.js]: Wonder if we should be excluding codeclimate from analyzing tests? I think their default config ignores tests. thoughts @searls ?

[the remark this answers is not in the document]
Comment (searls) [reply lib/log.test.js]: ¯\_(ツ)_/¯ it's your thingy man, whatever you want!

**Comment by jasonkarns** [comment]
Sample outputs given a `script/foo` which prints one line to STDOUT and one line to STDERR:

Things to note about this output:
- The script about to execute is printed at INFO level with one blank line before and after itself. This separates it from npm's preamble as well as separates from the script's own output (or script body in verbose mode)
- ALL scripty messages are prefixed with `scripty `, followed by a leader – `> ` for VERBOSE and INFO, `WARN ` for WARN, and `ERR!` for ERROR. The latter two copy npm directly. (No colorization yet.)
- Script body is printed only in VERBOSE mode. (with all lines prefixed) This means, by default, scripty only prints the script "preamble" (script name about to execute), and the script's own output.
- DryRun just adds one additional line about being in dry run mode (at INFO level)
- script failure prints the name of the script that failed, and the exit status at ERROR level
- "unable to run script": prints WARN level message about read/exec bit and continues (it _may_ find other scripts to exec, but most likely will be followed with "no script found error"
- "no script found" is printed at ERROR level, and splits the globs 1-per line to more easily see the list

<details><summary>Default configuration</summary>

```
$ npm run foo

> @nodenv/node-build@4.2.0 foo /Users/jasonkarns/Projects/nodenv/node-build
> scripty

scripty > Executing "/Users/jasonkarns/Projects/nodenv/node-build/script/foo":

Data to STDOUT
Diagnostics to STDERR
```
</details>

<details><summary>Verbose</summary>

```
$ SCRIPTY_VERBOSE=true npm run foo

> @nodenv/node-build@4.2.0 foo /Users/jasonkarns/Projects/nodenv/node-build
> scripty

scripty > Executing "/Users/jasonkarns/Projects/nodenv/node-build/script/foo":

scripty > #!/bin/bash
scripty > 
scripty > echo "Data to STDOUT"
scripty > echo "Diagnostics to STDERR" >&2

Data to STDOUT
Diagnostics to STDERR
```
</details>
<details><summary>Silent</summary>

```
$ SCRIPTY_SILENT=true npm run foo

> @nodenv/node-build@4.2.0 foo /Users/jasonkarns/Projects/nodenv/node-build
> scripty

Data to STDOUT
Diagnostics to STDERR
```
</details>
<details><summary>Silent + npm silent</summary>

```
$ SCRIPTY_SILENT=true npm -s run foo
Data to STDOUT
Diagnostics to STDERR
```
</details>
<details><summary>Dry Run</summary>

```
$ SCRIPTY_DRY_RUN=true npm run foo

> @nodenv/node-build@4.2.0 foo /Users/jasonkarns/Projects/nodenv/node-build
> scripty

scripty > This is a dry run. Executed scripts would be:

scripty > Executing "/Users/jasonkarns/Projects/nodenv/node-build/script/foo":

```
</details>
<details><summary>Default with script fail</summary>

```
$ npm run foo

> @nodenv/node-build@4.2.0 foo /Users/jasonkarns/Projects/nodenv/node-build
> scripty

scripty > Executing "/Users/jasonkarns/Projects/nodenv/node-build/script/foo":

Data to STDOUT
Diagnostics to STDERR
scripty ERR! script failed: '/Users/jasonkarns/Projects/nodenv/node-build/script/foo'
scripty ERR! exit status: 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @nodenv/node-build@4.2.0 foo: `scripty`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @nodenv/node-build@4.2.0 foo script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /usr/local/cache/npm/_logs/2018-12-21T02_48_04_397Z-debug.log
```
</details>
<details><summary>Default with un-executable script (or executable but not readable)</summary>

```
$ npm run foo

> @nodenv/node-build@4.2.0 foo /Users/jasonkarns/Projects/nodenv/node-build
> scripty

scripty WARN Ignoring script '/Users/jasonkarns/Projects/nodenv/node-build/script/foo' because it was not readable/executable.
scripty WARN Run `chmod u+rx '/Users/jasonkarns/Projects/nodenv/node-build/script/foo'` if you want scripty to run it.
scripty ERR! No script found for npm lifecycle 'foo' matching any of:
scripty ERR!   /Users/jasonkarns/Projects/nodenv/node-build/script/foo+(|.*)
scripty ERR!   /Users/jasonkarns/Projects/nodenv/node-build/script/foo/index+(|.*)
scripty ERR!   /Users/jasonkarns/Projects/nodenv/node-build/script/foo/*
scripty ERR!   /Users/jasonkarns/Projects/scripty/scripts/foo+(|.*)
scripty ERR!   /Users/jasonkarns/Projects/scripty/scripts/foo/index+(|.*)
scripty ERR!   /Users/jasonkarns/Projects/scripty/scripts/foo/*
scripty ERR! Either define a script or remove "scripty" from 'scripts.foo' in your package.json.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @nodenv/node-build@4.2.0 foo: `scripty`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @nodenv/node-build@4.2.0 foo script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /usr/local/cache/npm/_logs/2018-12-21T02_49_47_988Z-debug.log
```
</details>

**Comment by jasonkarns** [comment]
Merging as-is for now, will release as pre 2.0 and let it bake a bit. (dogfooding)

**Pull request merged**